src/rpc/mod.rs

```diff
@@ -544,42 +540,45 @@ impl Connection {
         let rx = self.send(req, channel_id)?;
 
         if payload.timeout_millis() == 0 {
-            // no-wait request,return Ok directly
+            // no-wait request,return Ok directly2
             return Ok(());
         }
 
-        let resp = match rx.recv_timeout(timeout) {
-            Ok(resp) => {
-                self.on_recv_in_time();
-                resp.map_err(|e| {
-                    error!(
+        // TODO: remove block_on with rx.await
+        let resp = self.runtimes.default_runtime.block_on(async move {
+            match TokioTimeout(timeout, rx).await {
```

Comment:
When use a free function from a module, just use `<module_name>::<function_name>`:
```suggestion
            match tokio::time::timeout(timeout, rx).await {
```